`Delete old Cabal build cache` step doesn't work when run from a fork

As seen in [this failing CI log](https://github.com/GaloisInc/crucible/actions/runs/15592156075/job/43913656909?pr=1423), the [`Delete old Cabal build cache`](https://github.com/GaloisInc/.github/blob/e08bfe4313abe456fe3792b072b05a51c0c513aa/.github/workflows/haskell-ci.yml#L352) step doesn't work when run from a PR that uses a fork:

```
 + gh cache delete -- 1-go-haskell-ci-cabal-build-ubuntu-24.04-cabal3.14.2.0-ghc9.6.7-refs/pull/1423/merge
X Failed to delete cache: HTTP 403: Resource not accessible by integration (https://api.github.com/repos/GaloisInc/crucible/actions/caches?key=1-go-haskell-ci-cabal-build-ubuntu-24.04-cabal3.14.2.0-ghc9.6.7-refs%2Fpull%2F1423%2Fmerge)
Error: Process completed with exit code 1.
```

Perhaps this should be skipped on forks?

--- a/.github/workflows/haskell-ci.yml
+++ b/.github/workflows/haskell-ci.yml
@@ -348,7 +348,9 @@ jobs:
       env:
         GH_TOKEN: ${{ github.token }}
       # See Note [Cache key check]
-      if: steps.build-cache.outputs.cache-hit && steps.build-cache.outputs.cache-primary-key == steps.build-cache.outputs.cache-matched-key
+      if: steps.build-cache.outputs.cache-hit &&
+          steps.build-cache.outputs.cache-primary-key == steps.build-cache.outputs.cache-matched-key &&
+          github.event.pull_request.head.repo.fork == false
       run: gh cache delete -- ${{ steps.build-cache.outputs.cache-primary-key }}
     - name: Save Cabal build cache
       uses: actions/cache/save@v4

